include/robot_state_publisher/robot_state_publisher.hpp

```diff
@@ -51,56 +51,117 @@ using MimicMap = std::map<std::string, urdf::JointMimicSharedPtr>;
 
-  KDL::Segment segment;
-  std::string root;
-  std::string tip;
+  KDL::Segment segment;  ///< The KDL segment
+  std::string root;  ///< The name of the root element to which this link is attached
+  std::string tip;  ///< The name of the element
 };
 
+/// The class that contains all of the functionality of the RobotStatePublisher ROS 2 node.
 class RobotStatePublisher : public rclcpp::Node
 {
 public:
   /// Constructor
   explicit RobotStatePublisher(const rclcpp::NodeOptions & options);
 
 protected:
-  /** Publish transforms to tf
-   * \param joint_positions A map of joint names and joint positions.
-   * \param time The time at which the joint positions were recorded
+  /** Setup the URDF for use.
+   * This method first parses the URDF into an internal representation.
+   * Based on that representation, it then generates the list of joint segments
+   * and mimic pairs that it needs during runtime.  Finally, it publishes
+   * the text of the URDF to the network on the /robot_description topic.
+   *
+   * \param[in] urdf_xml The string representing the URDF XML.
+   */
+  void setupURDF(const std::string & urdf_xml);
+
+  /** Recursive method to add all children to the internal segment list.
+   *
+   * \param[in] segment An iterator to the SegmentMap to add to the internal segment list.
+   */
+  void addChildren(const KDL::SegmentMap::const_iterator segment);
+
+  /** Publish transforms to /tf2.
+   * This method is called by callbackJointState() when new transforms are available and need to be published.
+   *
+   * \param[in] joint_positions A map of joint names to joint positions.
+   * \param[in] time The time at which the joint positions were recorded.
    */
   void publishTransforms(
     const std::map<std::string, double> & joint_positions,
     const builtin_interfaces::msg::Time & time);
 
+  /** Publish fixed transforms at startup time to /tf2_static.
+   */
   void publishFixedTransforms();
-  void addChildren(const KDL::SegmentMap::const_iterator segment);
+
+  /** The callback that is called when a new JointState message is received.
+   * This method examines the incoming JointStates and applies a series of checks to
+   * see if new transforms should be published.  If so, it calls publishTransforms() to do so.
+   *
+   * \param[in] state The JointState message that was delivered.
+   */
   void callbackJointState(const sensor_msgs::msg::JointState::SharedPtr state);
-  void setupURDF(const std::string & urdf_xml);
+
+  /** The callback that is called when parameter on the node are changed.
+   * This allows the class to dynamically react to changes in parameters.
+   *
+   * \param[in] parameters The list of parameters that are going to change.
+   */
   rcl_interfaces::msg::SetParametersResult parameterUpdate(
     const std::vector<rclcpp::Parameter> & parameters);
 
   std::map<std::string, SegmentPair> segments_;
+  ///< A map of dynamic segment names to SegmentPair structures
```

Comment:
Subjectively, I find it easier to read documentation before the member than after, but I wouldn't block on that.

Reply:
Same.

Reply:
I also agree.  The problem is that rosdoc2 didn't like that.  If I put the comment before the variable, then they would all be "shifted" down by 1 (e.g. it would say "A map of dynamic segment names to SegmentPair structures" for the `segments_fixed_` variable).

Let me poke at it again briefly and see if I can figure out what is going on there.

Reply:
Ah, it turns out I didn't understand Doxygen syntax.

```
///< document the member that comes *before* the text
```

```
/// document the member that comes *after* the text
```

I switched to the latter now, and it all seems happier.